protocols/kad/src/behaviour.rs

```diff
@@ -222,6 +222,13 @@ where
         Self::with_config(id, store, Default::default())
     }
 
+    /// Get the protocol name of this kademlia instance.
+    pub fn protocol_name(&self) -> Cow<'static, [u8]> {
```

Comment:
This API is a bit weird. If it's the default, then we return a `&'static [u8]`, and if it's not the default, we clone a `Vec<u8>`.
A more proper way to me would be to return a reference in both cases. In other words an enum of either `&'static [u8]` or `&'a [u8]`.
But doing some seems a bit too cumbersome, and in the end returning `&[u8]` looks like the best approach to me.